homeassistant/helpers/device_registry.py

```diff
@@ -1089,6 +1089,33 @@ def async_clear_label_id(self, label_id: str) -> None:
             self.async_update_device(device.id, labels=device.labels - {label_id})
 
 
+@callback
+async def async_remove_stale_device_links_helpers(
+    hass: HomeAssistant, entry_id: str, source_entity_id: str
+) -> None:
+    """Remove obsolete devices from helper config entries that inherit from source entity."""
+
+    def device_id_to_entity_id(source_entity_id: str) -> str | None:
+        """Resolve the device id to the entity id."""
+        # pylint: disable-next=import-outside-toplevel
+        from . import entity_registry
```

Comment:
We cant have this.

Reply:
I used it because there is already a similar use in this file. Do you think it's better to import at the beginning of the file?

https://github.com/home-assistant/core/blob/59ca5b04fa756952aed4a822605ec0435a1cb49c/homeassistant/helpers/device_registry.py#L1210-L1215

https://github.com/home-assistant/core/blob/59ca5b04fa756952aed4a822605ec0435a1cb49c/homeassistant/helpers/device_registry.py#L840-L850

Reply:
Imports are file io so its blocking event loop.

Reply:
If I move the import to the top of the file it causes circular import error with `entity_registry.py`. Can you give me a tip on the correct way in this case?
```
homeassistant/helpers/device_registry.py:31: in <module>
    from . import entity_registry, storage, translation
homeassistant/helpers/entity_registry.py:57: in <module>
    from .device_registry import (
E   ImportError: cannot import name 'EVENT_DEVICE_REGISTRY_UPDATED' from partially initialized module 'homeassistant.helpers.device_registry' (most likely due to a circular import) (/workspaces/core/homeassistant/helpers/device_registry.py)
```

Reply:
Move it to a different file

Reply:
Moved to a new file: `device_utils.py`